projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt

```diff
@@ -57,7 +59,68 @@ class CommunityApiController(
             log.error("Exception thrown when calling ${communityApiUrl + request.requestURI}. community-api returned ${ex.message}")
             ResponseEntity.status(ex.statusCode)
                 .headers(ex.responseHeaders)
-                .body(ex.responseBodyAsString);
+                .body(ex.responseBodyAsString)
+        }
+    }
+
+    @PostMapping("/compare")
+    fun compare(@RequestBody compare: Compare, request: HttpServletRequest): CompareReport {
+        val headers = request.headerNames.asSequence().associateWith { request.getHeader(it) }
+            .filter { it.key.lowercase() == "Authorization".lowercase() }.toMutableMap()
+
+        headers.forEach { log.info("Header name: ${it.key}") }
+
+        val uri = try {
+            Uri.valueOf(compare.uri)
+        } catch (ex: Exception) {
+            return compare.toReport("This endpoint is not configured")
+        }
+        val ccdJsonString = try {
+            compareService.toJsonString(compare)
+        } catch (ex: Exception) {
+            return compare.toReport("${uri.ccdFunction} bean cannot be found. Has this been implemented yet?")
+        }
+        val comApiUri = uri.comApiUrl.replace("{crn}", compare.crn)
+        val comApiJsonString = try {
+            communityApiClient.proxy(URI.create(communityApiUrl + comApiUri), headers).body!!
+        } catch (ex: HttpStatusCodeException) {
+            log.error("Exception thrown when calling ${communityApiUrl + comApiUri}. community-api returned ${ex.message}")
+            return compare.toReport(ex.message ?: "No message")
         }
+        val ccdJson = Json.createReader(StringReader(ccdJsonString)).readValue().asJsonObject()
+        val comApiJson = Json.createReader(StringReader(comApiJsonString)).readValue().asJsonObject()
+        val diff: JsonPatch = Json.createDiff(ccdJson, comApiJson)
+
+        val results = diff.toJsonArray().asSequence().associateWith {
+            val op = it.asJsonObject()["op"].toString()
+            val path = it.asJsonObject()["path"].toString()
+            if (op.contains("replace")) {
+                val ccdApiValue = getValueFromJsonPath(ccdJson, path)
+                "Values differ at ${it.asJsonObject()["path"].toString()} \n" +
+                    "Found $ccdApiValue in new API, but is ${it.asJsonObject()["value"].toString()} in Community API"
+            } else if (op.contains("remove")) {
+                "Additional element at $path exists in new API but is not present in Community API"
+            } else if (op.contains("add")) {
+                "Element at $path not found in new API, but is present with value of ${it.asJsonObject()["value"].toString()} in Community API "
+            } else {
+                "Unhandled operation $op"
+            }
+        }.map { x -> x.value }.toList()
```

Comment:
I don't think you need the associateWith here, as you're mapping it back to the value?  Could be a simple map:
```suggestion
        val results = diff.toJsonArray().map {
            val op = it.asJsonObject()["op"].toString()
            val path = it.asJsonObject()["path"].toString()
            if (op.contains("replace")) {
                val ccdApiValue = getValueFromJsonPath(ccdJson, path)
                "Values differ at $path \n" +
                    "Found $ccdApiValue in new API, but is ${it.asJsonObject()["value"].toString()} in Community API"
            } else if (op.contains("remove")) {
                "Additional element at $path exists in new API but is not present in Community API"
            } else if (op.contains("add")) {
                "Element at $path not found in new API, but is present with value of ${it.asJsonObject()["value"].toString()} in Community API "
            } else {
                "Unhandled operation $op"
            }
        }
```

Reply:
fixed during refactor